cpp/src/text/stemmer.cu

```diff
@@ -0,0 +1,256 @@
+ */
+struct dispatch_is_letter_fn {
+  template <typename T, std::enable_if_t<cudf::is_index_type<T>()>* = nullptr>
+  std::unique_ptr<cudf::column> operator()(cudf::strings_column_view const& strings,
+                                           letter_type ltype,
+                                           cudf::column_view const& indices,
+                                           rmm::mr::device_memory_resource* mr) const
+  {
+    CUDF_EXPECTS(strings.size() == indices.size(),
+                 "strings column and indices column must be the same size");
+    CUDF_EXPECTS(!indices.has_nulls(), "indices column must not contain nulls");
+    // resolve and pass an iterator for the indices column to the detail function
+    return is_letter(strings, ltype, indices.begin<T>(), 0, mr);
+  }
+  template <typename T, typename... Args, std::enable_if_t<not cudf::is_index_type<T>()>* = nullptr>
+  std::unique_ptr<cudf::column> operator()(Args&&... args) const
+  {
+    CUDF_FAIL("The is_letter indices parameter must be an integer type.");
+  }
+};
+
+}  // namespace
+
+std::unique_ptr<cudf::column> porter_stemmer_measure(cudf::strings_column_view const& strings,
+                                                     cudaStream_t stream,
+                                                     rmm::mr::device_memory_resource* mr)
+{
+  if (strings.size() == 0) return cudf::make_empty_column(cudf::data_type{cudf::type_id::INT32});
+
+  // create empty output column
+  auto results = cudf::make_fixed_width_column(cudf::data_type{cudf::type_id::INT32},
+                                               strings.size(),
+                                               copy_bitmask(strings.parent(), stream, mr),
+                                               strings.null_count(),
+                                               stream,
+                                               mr);
+  // compute measures into output column
+  auto strings_column = cudf::column_device_view::create(strings.parent(), stream);
+  thrust::transform(rmm::exec_policy(stream)->on(stream),
+                    thrust::make_counting_iterator<cudf::size_type>(0),
+                    thrust::make_counting_iterator<cudf::size_type>(strings.size()),
+                    results->mutable_view().data<int32_t>(),
+                    porter_stemmer_measure_fn{*strings_column});
+  return results;
+}
+
+}  // namespace detail
+
+// external APIs
+
+std::unique_ptr<cudf::column> is_letter(cudf::strings_column_view const& strings,
+                                        letter_type ltype,
+                                        cudf::size_type character_index,
+                                        rmm::mr::device_memory_resource* mr)
+{
+  CUDF_FUNC_RANGE();
+  return detail::is_letter(
+    strings, ltype, thrust::make_constant_iterator<cudf::size_type>(character_index), 0, mr);
+}
+
+std::unique_ptr<cudf::column> is_letter(cudf::strings_column_view const& strings,
```

Comment:
Shouldn't we have detail version of this ?

Reply:
You are right. The (non-existent) detail method should be doing the type-dispatch instead of here.
That is also why the stream is missing. :-)